app/org/sagebionetworks/bridge/services/HealthDataService.java

```diff
@@ -397,18 +400,42 @@ public HealthDataRecord getRecordById(String id) {
         return healthDataDao.getRecordsForUploadDate(uploadDate);
     }
 
-    public List<HealthDataRecord> getRecordsByHealthcodeCreatedOnSchemaId(String healthCode, Long createdOn, String schemaId) {
+    /** Gets a list of records for the given healthCode between the specified createdOn times (inclusive). */
+    public List<HealthDataRecord> getRecordsByHealthCodeCreatedOn(String healthCode, DateTime createdOnStart,
+            DateTime createdOnEnd) {
+        if (StringUtils.isBlank(healthCode)) {
+            throw new BadRequestException(String.format(Validate.CANNOT_BE_BLANK, "healthCode"));
+        }
```

Comment:
I think this should be a preconditions check. It should never be null if the system is working and it's not a 400 error to the user, it's a 500 error on us